### PR TITLE
[F2F-583]Mock | GOV.UK Notify | Mocking Gov Notify for emails

### PR DESCRIPTION
**Pull request by hemavana**
### What changed

Added a stub for GovNotify service which mocks the GOV.UK Notify POST /v2/notifications/email endpoint and responds with success or error responses in Dev, Build and Staging environments.
Evidences for all the responses have been attached to the ticket for reference.

### Why did it change

Enable Performance testing on F2F/IPR services to avoid hitting the GovNotify's test environment.

### Issue tracking

- [F2F-583](https://govukverify.atlassian.net/browse/F2F-583)

### PII logging

- [x] Verified that no PII data is being logged

### Environment variables or secrets

- Added new env var GOVUKNOTIFY_API to provide the stub url for dev/build/staging environments and use the GovNotify's prod url in Int/prod environments.

[F2F-583]: https://govukverify.atlassian.net/browse/F2F-583?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: gov-notify-stub/gov-notify-spec.yaml [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+      type: apiKey
+      name: X-Yoti-Auth-Digest
+      in: header
+      description: "Signed request performed by back-end SDKs that includes a `X-Yoti-Auth-Digest`"
+    X-Yoti-Auth-Token:
+      type: apiKey
+      name: X-Yoti-Auth-Token
+      in: header
+      description: "`X-Yoti-Auth-Token` header containing the `client_session_token` value obtained via the `Session` creation endpoint."
+
```
Comment (darrendunford) [inline gov-notify-stub/gov-notify-spec.yaml:186]: do we need this bit?

File: gov-notify-stub/gov-notify-spec.yaml [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+  ## ---------------------
+  ## Example Definitions
+  ## ---------------------
+  examples:
```
Comment (darrendunford) [inline gov-notify-stub/gov-notify-spec.yaml:190]: are these examples all relevant to the GovNotify stub?

**Review by darrendunford — DISMISSED** [review]

File: gov-notify-stub/gov-notify-spec.yaml (hunk 1 above)
Comment (hemavana) [reply gov-notify-stub/gov-notify-spec.yaml]: Not required, Removed the unwanted definitions and schemas.

File: gov-notify-stub/gov-notify-spec.yaml (hunk 2 above)
Comment (hemavana) [reply gov-notify-stub/gov-notify-spec.yaml:160]: Not required, Removed the unwanted definitions and schemas.

**Review by darrendunford — DISMISSED** [review]

**Review by darrendunford — DISMISSED** [review]

File: src/GovNotifyHandler.ts [hunk 3]
```diff
@@ -46,7 +46,8 @@ class GovNotifyHandler implements LambdaInterface {
 						throw err;
 					}
 				}
-				await SendEmailProcessor.getInstance(logger, metrics, GOVUKNOTIFY_API_KEY, this.environmentVariables.sessionEventsTable()).processRequest(body);
+				const govnotifyServiceId = GOVUKNOTIFY_API_KEY.substring(GOVUKNOTIFY_API_KEY.length - 73, GOVUKNOTIFY_API_KEY.length - 37);
+				await SendEmailProcessor.getInstance(logger, metrics, GOVUKNOTIFY_API_KEY, govnotifyServiceId, this.environmentVariables.sessionEventsTable()).processRequest(body);
```
Comment (richa-misra) [inline src/GovNotifyHandler.ts:50]: could we wrap it in a try/catch pls incase substring throws an exception and call SendEmailProcessor only if  govnotifyServiceId is truthy

File: deploy/template.yaml [hunk 4]
```diff
@@ -122,6 +124,7 @@ Mappings:
       RETURNREDIRECTURL: "https://return.build.account.gov.uk/callback"
     staging:
       GOVUKNOTIFYTEMPLATEID: "3baf3b59-2396-44c8-b7a4-b5a981fad5e8"
+      GOVUKNOTIFYAPI: "https://ipr-govnotify-stub-govnotifystub.return.staging.account.gov.uk/govnotify"
```
Comment (richa-misra) [inline deploy/template.yaml:127]: planning for notify stub on staging?

File: src/GovNotifyHandler.ts (hunk 3 above)
Comment (hemavana) [reply src/GovNotifyHandler.ts:57]: Done
